content/blog/2019/08/2019-08-13-remoting-kafka-kubernetes-release-2.0.adoc

```diff
@@ -0,0 +1,92 @@
+
+== Dynamically provision Remoting Kafka Agent in Kubernetes
+
+With previous version, users have to manually add/remove nodes so it is hard to scale builds quickly. link:https://github.com/jenkinsci/kubernetes-plugin[Kubernetes plugin] allows us to dynamically provision agents in Kubernetes but it is designed for JLPT agent. With this new version, Remoting Kafka agent can also be provisioned automatically in Kubernetes environment.
+
+image:/images/post-images/remoting-kafka/cloud-ui.png[Remoting Kafka Cloud UI]
+
+Users can find the new feature in Cloud section. Here users could input Kubernetes connection parameters and desired Remoting Kafka agent properties including labels. When new build with matching labels get started and there is not any free nodes, Cloud will automatically provision Remoting Kafka agent pod in Kubernetes to run the build.
+
+image:/images/post-images/remoting-kafka/cloud-nodes.png[Remoting Kafka Agent get provisioned]
+
+== Helm Chart
+
+Helm chart for Remoting over Apache Kafka plugin is based on link:https://github.com/helm/charts/tree/master/stable/jenkins[stable/jenkins] chart and link:https://github.com/helm/charts/tree/master/incubator/kafka[incubator/kafka] chart. You can follow the instruction link:https://github.com/jenkinsci/remoting-kafka-plugin/blob/master/README.md[here] to install a demo ready-to-use Helm release, your `kubectl get all` should look like this:
+
+[source, bash]
+----
+NAME                                READY   STATUS    RESTARTS   AGE
+pod/demo-jenkins-64dbd87987-bmndf   1/1     Running   0          2m21s
+pod/demo-kafka-0                    1/1     Running   0          2m21s
+pod/demo-zookeeper-0                1/1     Running   0          2m21s
+
+NAME                              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+service/demo-jenkins              NodePort    10.108.238.56   <none>        8080:30386/TCP               2m21s
+service/demo-jenkins-agent        ClusterIP   10.98.85.184    <none>        50000/TCP                    2m21s
+service/demo-kafka                ClusterIP   10.109.231.58   <none>        9092/TCP                     2m21s
+service/demo-kafka-headless       ClusterIP   None            <none>        9092/TCP                     2m21s
+service/demo-zookeeper            ClusterIP   10.103.2.231    <none>        2181/TCP                     2m21s
+service/demo-zookeeper-headless   ClusterIP   None            <none>        2181/TCP,3888/TCP,2888/TCP   2m21s
+
+NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/demo-jenkins   1/1     1            1           2m21s
+
+NAME                                      DESIRED   CURRENT   READY   AGE
+replicaset.apps/demo-jenkins-64dbd87987   1         1         1       2m21s
+
+NAME                              READY   AGE
+statefulset.apps/demo-kafka       1/1     2m21s
+statefulset.apps/demo-zookeeper   1/1     2m21s
+----
+
+== Next Phase Plan
```

Comment:
Mazbe not needed. We are in the middle of phase 3, and it can be rather a confusion